utils/i18nUtil.js

```diff
@@ -0,0 +1,20 @@
+import i18n from "i18next";
+import { initReactI18next } from "react-i18next";
+import en from "../locales/en.json";
+import fr from "../locales/fr.json";
```

Comment:
Do those end up being part of the output bundle? If so, have we considered creating different bundles such that we can dynamically pull the one needed based on the language of the admin?

Reply:
Yes; I took a very simplistic approach in this PR, because I wasn't sure what approach & how opinionated we wanted to get on it (i.e. demonstrate "best practices", or let the developer make those decisions as their needs change?)

Full disclosure, my background is mostly in back end development, and I don't fully understand the concepts of bundling just yet.

There's a few approaches we could take with this:
- I [protoyped](https://github.com/Shopify/trishrempel-app-react-i18next/blob/main/web/frontend/i18n.js) using a [Http Backend plugin](https://github.com/i18next/i18next-http-backend), which requires putting the locale files in the `public` folder. The application then dynamically loads translation files as needed (and you can optionally select to pre-load English by default, for example). I _think_ this would mean that the JSON locales would not be included in the output bundle, but it does mean that the React `Suspense` component is required to display a loading state while translations are loaded.
- Dynamically load translations ourselves as locale is switched. @movermeyer [prototyped this](https://github.com/Shopify/movermeyer-app-react-intl/blob/main/web/frontend/components/providers/AsyncI18nProvider.jsx) by asynchronously importing language files on locale selection.
- Other - there are a lot of other [backend plugins](https://www.i18next.com/overview/plugins-and-utils#backends) available that could help optimize the build, though I quickly got lost trying to figure out what a good but simple "best practices" solution might look like.

Reply:
I added a commit that uses the [Http Backend plugin](https://github.com/i18next/i18next-http-backend) - this requires the translations to be in the `public` folder to be retrieved as assets. This will require minor changes to the Ruby and PHP templates to copy and link the `public/locales` folder on build like the `public/assets` folder.

Reply:
I've updated this PR to use [`i18next-resources-to-backend`](https://github.com/i18next/i18next-resources-to-backend) and wait for the application locales and Polaris locales to be dynamically loaded before rendering React, and it no longer requires a `Suspense` component.